cache/cache.go

```diff
@@ -63,6 +63,7 @@ type Shard struct {
 	items            map[string]*points.Points
 	notConfirmed     []*points.Points // linear search for value/slot
 	notConfirmedUsed int              // search value in notConfirmed[:notConfirmedUsed]
+	adds             map[string]struct{} // map to maintain all the new metric names
```

Comment:
not really about this PR, but if we could move to lock free map or just sync.Map for items. I guess the memory overhead from cacheMetrics in fileListUpdater could be saved.